src/main/java/org/broadinstitute/consent/http/service/UserService.java

```diff
@@ -342,6 +368,29 @@ public void insertUserRoles(List<UserRole> roles, Integer userId) {
         userRoleDAO.insertUserRoles(roles, userId);
     }
 
+    private Integer getSigningOfficialForUser(Integer userId) {
+        List<UserProperty> props =
+                userPropertyDAO.findUserPropertiesByUserIdAndPropertyKeys(
+                        userId,
+                        List.of(UserFields.SELECTED_SIGNING_OFFICIAL_ID.getValue())
+                );
+
+        if (props.size() == 0) {
+            return null;
```

Comment:
I generally try to avoid using null as a state (and am guilty of doing this myself). If we find ourselves needing it, try returning `Optional<T>` instead so that callers can know that the method may or may not have a valid value.